torch/_dynamo/symbolic_convert.py

```diff
@@ -2121,6 +2124,19 @@ def __init__(
                 if name in f_locals:
                     self._freevars_ids[name] = id(f_locals[name])
 
+    def _check_for_optimizer_step_closure(self):
+        from .variables import OptimizerVariable
+
+        if (
+            "closure" in self.symbolic_locals
+            and not isinstance(self.symbolic_locals["closure"], ConstantVariable)
+            and "self" in self.symbolic_locals
+            and isinstance(self.symbolic_locals["self"], OptimizerVariable)
+            and isinstance(self.symbolic_locals["self"].value, torch.optim.Optimizer)
+            and self.code_options["co_name"] == "step"
+        ):
+            unimplemented("Optimizer step with closure not supported.")
+
```

Comment:
I'm curious if we can merge this and the changes at optimizer.py into the same place, it seems they do same thing. Can we define a rule at ```trace_rule.lookup_inner``` like if function is optimizer.step and first arg is closure, then graph break?

Reply:
yeah that sounds awesome! didn't know it was possible. I was pretty annoyed to have to include a case for "if we're tracing" and "if we are compiling that function"